console/console-init/ui/src/modules/iot-device/containers/EmptyDeviceContainer/EmptyDeviceContainer.tsx

```diff
@@ -42,12 +49,24 @@ export const EmptyDeviceContainer: React.FC<IEmptyDeviceContainerProps> = ({
 
   const total = data?.devices?.total || 0;
 
-  total > 0 && setTotalDevices(total);
+  useEffect(() => {
+    if (total === 0) {
+      setShowEmptyDevice(true);
+    } else if (total > 0) {
+      setShowEmptyDevice(false);
+    }
+  }, [total]);
```

Comment:
just passing total will create warning. Please resolve this warning if possible

Reply:
This is kind of common probelms in useEffect so we will have to find out the solution.